libraries/chain/block_log.cpp

```diff
@@ -1094,9 +1095,13 @@ namespace eosio { namespace chain {
          }
 
          signed_block_ptr retry_read_block_by_num(uint32_t block_num) final {
-            auto ds = catalog.ro_stream_for_block(block_num);
-            if (ds)
-               return read_block(*ds, block_num);
+            uint64_t block_size = 0;
+
+            auto ds = catalog.ro_stream_and_size_for_block(block_num, block_size);
```

Comment:
Not from your changes. I wish `ro_stream_and_size_for_block` returned `ds` and `size` explicitly.